packages/ui/src/components/UnitInput/index.tsx

```diff
@@ -0,0 +1,329 @@
+        </Stack>
+      ) : null}
+      <UnitInputWrapper
+        direction="row"
+        data-testid={dataTestId}
+        data-size={size}
+        data-focus={hasFocus}
+        width={width}
+        id={id}
+        data-success={!!success}
+        data-error={!!error}
+        onMouseEnter={() =>
+          !disabled && !readOnly ? setHovered(true) : undefined
+        }
+        onMouseLeave={() => setHovered(false)}
+        data-disabled={!!disabled}
+        data-readonly={!!readOnly}
+      >
+        <StyledInputWrapper id="input-field" data-hovered={hovered}>
+          <StyledInput
+            type="number"
+            aria-invalid={!!error}
+            autoFocus={autoFocus}
+            disabled={disabled}
+            data-readOnly={readOnly}
+            name={`${name}-value`}
+            value={val}
+            onChange={event => {
+              const numericValue = Number.parseInt(event.target.value, 10)
+              if (numericValue > maxValue) {
+                setVal(maxValue)
+                onChange?.(maxValue)
+              } else if (numericValue < minValue) {
+                setVal(minValue)
+                onChange?.(minValue)
+              } else {
+                setVal(numericValue)
+                onChange?.(numericValue)
+              }
+            }}
+            placeholder={placeholder}
+            max={maxValue}
+            readOnly={readOnly}
+            min={minValue}
+            step={step}
+            data-success={success}
+            data-error={error}
+            data-testid="unit-input"
+            required={required}
+            className={className}
+            onFocus={() =>
+              !disabled && !readOnly ? setHasFocus(true) : undefined
+            }
+          />
+          {error ? <Icon name="alert" sentiment="danger" /> : null}
+          {success && !error ? (
+            <Icon name="checkbox-circle-outline" sentiment="success" />
+          ) : null}
+        </StyledInputWrapper>
+        <CustomSelectInput
+          width={selectInputWidth}
+          id={`${name}-unit`}
+          name={`${name}-unit`}
+          onChange={newValue => {
+            onChangeUnitValue(newValue)
+            setHasFocus(false)
+            setHovered(false)
+          }}
+          error={unitError}
+          value={unitValue}
+          options={options}
+          searchable={false}
+          clearable={false}
+          placeholder={placeholderUnit}
+          disabled={disabled || options.length === 1}
+          size={size}
+          multiselect={false}
+          onFocus={() =>
+            !disabled && !readOnly ? setHasFocus(true) : undefined
+          }
+          readOnly={readOnly}
+          onBlur={() => setHasFocus(false)}
+        />
+      </UnitInputWrapper>
+      {helper ? (
+        <Text as="p" variant="caption" sentiment={sentiment}>
+          {helper}
+        </Text>
+      ) : null}
```

Comment:
And so here you can also follow what has been done with other inputs and success, error props